devices/giex.js

```diff
@@ -1,135 +1,203 @@
-                return {state: {cycle_irrigation_interval: value}};
+            case keys.giexWaterValve.mode:
+                await tuya.sendDataPointBool(entity, dataPoints.giexWaterValve.mode, value === CAPACITY);
+                return {state: {[keys.giexWaterValve.mode]: value}};
+            case keys.giexWaterValve.irrigationTarget:
+                await tuya.sendDataPointValue(entity, dataPoints.giexWaterValve.irrigationTarget, value);
+                return {state: {[keys.giexWaterValve.irrigationTarget]: value}};
+            case keys.giexWaterValve.cycleIrrigationNumTimes:
+                await tuya.sendDataPointValue(entity, dataPoints.giexWaterValve.cycleIrrigationNumTimes, value);
+                return {state: {[keys.giexWaterValve.cycleIrrigationNumTimes]: value}};
+            case keys.giexWaterValve.cycleIrrigationInterval:
+                await tuya.sendDataPointValue(entity, dataPoints.giexWaterValve.cycleIrrigationInterval, value);
+                return {state: {[keys.giexWaterValve.cycleIrrigationInterval]: value}};
             default: // Unknown key warning
-                meta.logger.warn(`tz.GIEX_water_valve: Unhandled key ${key}`);
+                meta.logger.warn(`tzLocal.giexWaterValve: Unhandled KEY ${key}`);
             }
         },
     },
 };
 
+const exportTemplates = {
+    giexWaterValve: {
+        vendor: 'GiEX',
+        description: 'Water irrigation valve',
+        onEvent: tuya.onEventSetLocalTime,
+        toZigbee: [tzLocal.giexWaterValve],
+        exposes: [
+            ep.battery(),
+            exposes.binary(keys.giexWaterValve.state, ea.STATE_SET, ON, OFF)
+                .withDescription('State'),
+            exposes.enum(keys.giexWaterValve.mode, ea.STATE_SET, [DURATION, CAPACITY])
+                .withDescription('Irrigation mode'),
+            exposes.numeric(keys.giexWaterValve.cycleIrrigationNumTimes, ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(100)
+                .withDescription('Number of cycle irrigation times, set to 0 for single cycle'),
+            exposes.numeric(keys.giexWaterValve.irrigationStartTime, ea.STATE)
+                .withDescription('Last irrigation start time'),
+            exposes.numeric(keys.giexWaterValve.irrigationEndTime, ea.STATE)
+                .withDescription('Last irrigation end time'),
+            exposes.numeric(keys.giexWaterValve.lastIrrigationDuration, ea.STATE)
+                .withDescription('Last irrigation duration'),
+            exposes.numeric(keys.giexWaterValve.waterConsumed, ea.STATE)
+                .withUnit('L')
+                .withDescription('Last irrigation water consumption'),
+        ],
+    },
+};
+
 module.exports = [
+    // _TZE200_a7sghmms uses seconds, timezone is local
     {
+        ...exportTemplates.giexWaterValve,
+        model: 'QT06 (a7sghmms)',
```

Comment:
@Koenkk I have a conflict with the model name: if I reuse it then the `Verify definitions` test fails with a `Duplicate model QT06` error, but I if specify a different model name then the link to documentation is broken. Is there any elegant way to solve this? Or should we just add new documentation pages for these models? Since there are slight differences in behavior.

Reply:
I propose to name them `QT06_1` and `QT06_2`

Reply:
Ok, I'll push those names and open the related PR in zigbee2mqtt.io asap.